kernel/pipes.c

```diff
@@ -528,7 +528,18 @@ int z_pipe_put_internal(struct k_pipe *pipe, struct k_pipe_async *async_desc,
 		return 0;
 	}
 
-	/* Not all data was copied. */
+	if (K_TIMEOUT_EQ(timeout, K_FOREVER) && num_bytes_written >= min_xfer) {
```

Comment:
How about `if (!K_TIMEOUT_EQ(timeout, K_NO_WAIT))` instead?

Reply:
That could work. Will update.